src/components/search/SearchDirective.js

```diff
@@ -1,756 +1,259 @@
+  var ResultStats = function() {
+    this.results = {
+      'locations': -1,
+      'featuresearch': -1,
+      'layers': -1
+    };
+  };
+
+  ResultStats.prototype.reset = function() {
+    for (var type in this.results) {
+      this.results[type] = -1;
+    }
+  };
+
+  ResultStats.prototype.update = function(type, nr) {
+    this.results[type] = nr;
+  };
+
+  ResultStats.prototype.sets = function() {
+    var ret = 0;
+    for (var type in this.results) {
+      ret += this.results[type] > 0 ? 1 : 0;
+    }
+    return ret;
+  };
+
+  ResultStats.prototype.done = function() {
+    for (var type in this.results) {
+      if (this.results[type] < 0) {
+        return false;
+      }
+    }
+    return true;
+  };
+
+  ResultStats.prototype.totalResults = function() {
+    var ret = 0;
+    for (var type in this.results) {
+      ret += this.results[type] !== -1 ? this.results[type] : 0;
+    }
+    return ret;
+  };
+
+  var swisssearchActive = false;
+  var keepssParam = false;
+
+  module.controller('GaSearchDirectiveController',
+    function($scope, $rootScope, $translate, $sce, $timeout, gaPermalink,
+             gaUrlUtils, gaSearchGetCoordinate, gaMapUtils, gaMarkerOverlay,
+             gaKml) {
+      var blockQuery = false;
+      var restat = new ResultStats();
+      restat.reset();
+      $scope.restat = restat;
+
+      // Child options are used as a carrier of communication
+      // between main search directive and result directive
+      $scope.childoptions = {};
+
+      $scope.childoptions.featureUrl = $scope.options.featureUrl;
+
+      // Result set announces a result selection
+      $scope.childoptions.valueSelected = function(strValue) {
+        $scope.query = strValue;
+        blockQuery = true;
+
+        // Take care of swisssearch parameter
+        if (swisssearchActive) {
+          // If we end up here because swisssearch cause a single click, we
+          // remove the swisssearch parameter on move/zoom.
+          if (restat.totalResults() == 1) {
+            keepssParam = true;
+            var unregProp = $scope.map.getView().on('propertychange',
+                                             function() {
+              gaPermalink.deleteParam('swisssearch');
+              $scope.map.getView().unByKey(unregProp);
+            });
+          } else {
+            gaPermalink.deleteParam('swisssearch');
+          }
+          swisssearchActive = false;
+          $rootScope.$broadcast('gaSwisssearchDone');
         }
+      };
 
-        function zoomToExtent(map, extent) {
-          var size = map.getSize();
-          var view = map.getView();
+      $scope.childoptions.query = '';
 
-          view.fitExtent(extent, size);
+      $scope.clearInput = function() {
+        restat.reset();
+        gaMarkerOverlay.remove($scope.map);
+        $scope.query = '';
+        $scope.childoptions.query = '';
+        if ($scope.input && $scope.input.blur) {
+          $scope.input.blur();
```

Comment:
It's like that on current prod.

But I can understand your point, but would like to do it after this is merged. Please open an issue.